src/Authentication/Authentication.Core/Authenticator.cs

```diff
@@ -29,23 +30,51 @@ public static class Authenticator
         /// <param name="forceRefresh">Whether or not to force refresh a token if one exists.</param>
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns></returns>
-        public static async Task<IAuthContext> AuthenticateAsync(IAuthContext authContext, bool forceRefresh, CancellationToken cancellationToken)
+        public static Task<(IAuthContext context, AuthError authError)> AuthenticateAsync(IAuthContext authContext, bool forceRefresh, CancellationToken cancellationToken)
+        {
+            // Gets a static instance of IAuthenticationProvider when the client app hasn't changed.
+            var (authProvider, clientApplication, _) = GetClientApplication(authContext);
+            return AuthenticateAsync(clientApplication, authProvider, authContext, forceRefresh, cancellationToken);
+        }
+
+        /// <summary>
+        /// Authenticates the client using the provided <see cref="IAuthContext"/>.
+        /// </summary>
+        /// <param name="authContext">The <see cref="IAuthContext"/> to authenticate.</param>
+        /// <param name="forceRefresh">Whether or not to force refresh a token if one exists.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <param name="fallBackWarning"></param>
+        /// <returns></returns>
+        public static Task<(IAuthContext context, AuthError authError)> AuthenticateAsync(IAuthContext authContext, bool forceRefresh, CancellationToken cancellationToken, Action<AuthProviderType> fallBackWarning)
```

Comment:
This is not needed if `fallBackWarning` is made nullable in `AuthenticateAsync`. See my suggestion below.